components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayJMSMessageListener.java

```diff
@@ -55,6 +58,7 @@
     private InMemoryAPIDeployer inMemoryApiDeployer = new InMemoryAPIDeployer();
     GatewayArtifactSynchronizerProperties gatewayArtifactSynchronizerProperties = ServiceReferenceHolder
             .getInstance().getAPIManagerConfiguration().getGatewayArtifactSynchronizerProperties();
+    ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(10);
```

Comment:
As a practice give a name to Threads in the thread pool. So use the override of newScheduledThreadPool which accepts a ThreadFactory, and use it to give a name.